Zend/zend_builtin_functions.stub.php

```diff
@@ -2,6 +2,7 @@
 
 /** @generate-class-entries */
 
+#[\AllowDynamicProperties]
```

Comment:
I'm not sure if we want to use attributes here. For contrast, `ZEND_ACC_NO_DYNAMIC_PROPERTIES` is added via the `@strict-properties` PHPDoc annotation and I'd be fine with using the same mechanism.

Also, it would be nice to either:
- not to allow `ZEND_ACC_NO_DYNAMIC_PROPERTIES` and `ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES ` in the same type
- or refactor the annotation to only allow one value

Reply:
> I'm not sure if we want to use attributes here.

This PR is intended to test the waters for Attribute support in stubs with a simple case. I'm also planning to migrate `\SensitiveParameter` to actual attributes, because that feels much more natural for the stub writer and also for consumers that take the stubs as is.

> For contrast, ZEND_ACC_NO_DYNAMIC_PROPERTIES is added via the @strict-properties PHPDoc annotation and I'd be fine with using the same mechanism.

If I'm not mistaken that flag cannot be applied in userland code, right? In that case it makes sense to define that as PHPDoc to not pretend that it would work in userland. The `AllowDynamicProperties` attribute on the other hand can be used as-is in userland code.

> not to allow ZEND_ACC_NO_DYNAMIC_PROPERTIES and ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES in the same type

That makes sense. Would this be something you would add to the stub generator or should this actually be enforced in C? The latter would likely be above my pay grade with regard to Internals :smiley:

Reply:
> That makes sense. Would this be something you would add to the stub generator or should this actually be enforced in C? The latter would likely be above my pay grade with regard to Internals 

For me, only doing so in gen_stub.php is fine.

Reply:
Okay, updated to add the check.